cpo/lib/openshift/oc.py

```diff
@@ -226,3 +228,74 @@ def log_in_to_openshift_cluster_with_token(server: str, token: str):
     oc_login_args = get_oc_login_args_with_token(server, token)
 
     execute_oc_command(oc_login_args)
+
+
+def get_deployment_name(search_string: str) -> List[str]:
+    """Returns the available Openshift deployment(s) for a given search string"""
+
+    oc_get_deployments_args = ["get", "deployments"]
+
+    oc_get_deployments_result = execute_oc_command(oc_get_deployments_args, capture_output=True).stdout
+
+    result = []
+    for line in oc_get_deployments_result.splitlines():
+        if search_string in line:
+            result.append(line.split()[0].strip())
+
+    if not result:
+        raise DataGateCLIException(f"Deployment(s) containing the string '{search_string}' could not be found")
+
+    return result
+
+
+def get_pod_name(search_string: str) -> List[str]:
+    """Returns the available Openshift pod(s) for a given search string"""
+
+    oc_get_pods_args = ["get", "pods"]
+
+    oc_get_pods_result = execute_oc_command(oc_get_pods_args, capture_output=True).stdout
+
+    result = []
+    for line in oc_get_pods_result.splitlines():
+        if search_string in line:
+            result.append(line.split()[0].strip())
+
+    if not result:
+        raise DataGateCLIException(f"Pod(s) containing the string '{search_string}' could not be found")
+
+    return result
+
+
+def get_pod_status(pod_name: str) -> GetPodEntry:
+    """Returns the current pod status for a given pod name"""
+    oc_get_pods_args = ["get", "pods"]
```

Comment:
Either return and parse JSON (`oc get pods --output json`) or use JSONPath (`oc get pods {pod_name} --output jsonpath="{.status.phase}"`)